font/src/darwin/mod.rs

```diff
@@ -482,13 +481,13 @@ impl Font {
         }
 
         let glyph_index = self.glyph_index(character)
-            .ok_or(Error::MissingGlyph(character))?;
+            .ok_or_else(|| Error::MissingGlyph(character))?;
```

Comment:
:eyes:

Reply:
Do you have any issue with this change?

Reply:
No, I'm just kinda surprised.

Reply:
Ah okay, it's an interesting one:

`ok_or` is not lazy, so `Error::MissingGlpyh(character)` is always called, even if the result is `Ok`. So this should only be used for stuff like `32` where no expensive initialization is required.

However with `ok_or_else` this is done lazily, so `Error::MissingGlyph(character)` is only called if the result is `Err` and not okay, so we save ourselves the call if the result is `Ok`.

This is a pretty interesting lint imho because I think a lot of people don't know this difference. At least I was only made aware of it when I first saw the clippy lint, but it makes sense when you take a look at it knowing what it does.